packages/zoe/src/contractSupport/ratio.js

```diff
@@ -0,0 +1,135 @@
+import './types';
+import { assert, details as X, q } from '@agoric/assert';
+import { Nat } from '@agoric/nat';
+import { natSafeMath } from '.';
+
+const { multiply, floorDivide } = natSafeMath;
+
+// make a Ratio, which represents a fraction. It is a pass-by-copy record.
+//
+// The natural syntax for the most common operations we want to support
+// are Amount * Ratio and Amount / Ratio. Since the operations want to adhere to
+// the ratio rather than the amount, we settled on a calling convention of
+// multiplyBy(Amount, Ratio) and divideBy(Amount, Ratio).
+//
+// The most common kind of Ratio can be applied to Amounts of a particular
+// brand, and produces results of the same brand. This represents a multiplier
+// that is only applicable to that brand. The less common kind of Ratio can be
+// applied to one particular brand of amounts, and produces results of another
+// particular brand. This represents some kind of exchange rate. The
+// brand-checking helps us ensure that normal Ratios aren't applied to amounts
+// of the wrong brand, and that exchange rates are only used in the appropriate
+// direction.
+
+const PERCENT = 100n;
+
+const ratioPropertyNames = ['numerator', 'denominator'];
+
+export function assertIsRatio(ratio) {
```

Comment:
Small thing, but can we not use the `function` keyword for consistency with the rest of Zoe?